hedera-node/src/main/java/com/hedera/services/utils/BytesComparator.java

```diff
@@ -0,0 +1,57 @@
+package com.hedera.services.utils;
+
+import org.apache.tuweni.bytes.Bytes;
+
+import java.util.Comparator;
+
+/**
+ * Compares Bytes as if they are overly large unsingned integers.
+ */
+public class BytesComparator implements Comparator<Bytes> {
```

Comment:
TreeMap requires a Comparator - https://github.com/hashgraph/hedera-services/pull/2850/files#diff-c988aa8c938f1dcd40e15f90ecd6b5dd0a22c5b34d6d1ac8594eca43c403acd3R259

We use tree map because we want sorted iteration.

Reply:
> https://github.com/hashgraph/hedera-services/pull/2850/files#diff-c988aa8c938f1dcd40e15f90ecd6b5dd0a22c5b34d6d1ac8594eca43c403acd3R259

But if the key type already implements `Comparable<T>` (as with `Bytes`), the explicit comparator isn't needed, right?